xwiki-commons-core/xwiki-commons-xml/src/test/java/org/xwiki/xml/XMLUtilsTest.java

```diff
@@ -498,10 +498,11 @@ void transformAvoidXMLBomb()
 
         assertEquals(1, logCapture.size());
         assertEquals(Level.WARN, logCapture.getLogEvent(0).getLevel());
-        assertEquals("Failed to apply XSLT transformation: [javax.xml.transform.TransformerException: "
-            + "com.sun.org.apache.xml.internal.utils.WrappedRuntimeException: The parser has encountered more than "
-            + "\"100,000\" entity expansions in this document; this is the limit imposed by the application.]",
-            logCapture.getMessage(0));
+        String txt = "Failed to apply XSLT transformation: \\[javax.xml.transform.TransformerException: "
+                + "com.sun.org.apache.xml.internal.utils.WrappedRuntimeException: The parser has encountered more than "
+                + "\"100.000\" entity expansions in this document; this is the limit imposed by the application.]";
```

Comment:
This is missing a lot of regex escapes. Best to use the general escape chars `\Q` and `\E` for all parts except the number.

Reply:
changed the string with suggested escaping

Reply:
changed assertion to hamcrest